src/test_l3_dsCompositeIn.c

```diff
@@ -0,0 +1,565 @@
+
+    gSignalStatus = sigStatus;
+}
+
+/**
+ * @brief Callback function for CompositeIn status change.
+ *
+ * This function is invoked whenever a status change occurs in the CompositeIn connection.
+ */
+static void compositeInStatusChangeCB(dsCompositeInStatus_t inputStatus)
+{
+    UT_LOG_INFO("Received statuschange callback isPresented: %s, activeport: %s",
+                 ut_control_GetMapString(bool_mapTable, inputStatus.isPresented),
+                 ut_control_GetMapString(dsCompositeInPortMappingTable, inputStatus.activePort));
+
+    for(int i = 0 ; i < dsCOMPOSITE_IN_PORT_MAX ; i++) {
+
+         if(!(inputStatus.isPresented))
+                 continue;
+
+         UT_LOG_INFO("Received statuschange callback isPortConnected: %s, activeport: %s",
+                 ut_control_GetMapString(bool_mapTable, inputStatus.isPortConnected[i]),
+                 ut_control_GetMapString(dsCompositeInPortMappingTable, inputStatus.activePort));
+    }
+
+    gStatusChange = inputStatus;
+}
+
+/**
+* @brief This test initializes the CompositeIn Module.
+*
+* This test function initializes the CompositeIn Module.
+*
+* **Test Group ID:** 03@n
+* **Test Case ID:** 001@n
+*
+* **Test Procedure:**
+* Refer to Test specification documentation
+* [dsCompositeIn_L3_Low-Level_TestSpecification.md](../docs/pages/ds-compositeIn_L3_Low-Level_TestSpecification.md)
+*/
+void test_l3_CompositeIn_initialize(void)
+{
+    gTestID = 1;
+    UT_LOG_INFO("In %s [%02d%03d]", __FUNCTION__, gTestGroup, gTestID);
+
+    dsError_t ret = dsERR_NONE;
+
+    /* Initialize the CompositeIn Module */
+    UT_LOG_INFO("Calling dsCompositeInInit()");
+    ret = dsCompositeInInit();
+    if(ret != dsERR_NONE) {
+        UT_LOG_ERROR("Failed dsCompositeInInit() dsError_t:[%s]", ut_control_GetMapString(dsError_mapTable, ret));
+    }
+    else {
+        UT_LOG_INFO("Result dsCompositeInInit() dsError_t:[%s]", ut_control_GetMapString(dsError_mapTable, ret));
+    }
+
+    /* Register connection status callback */
+    UT_LOG_INFO("Calling dsCompositeInRegisterConnectCB(CBFunc:[0x%0X])", compositeInConnectCB);
+    ret = dsCompositeInRegisterConnectCB(compositeInConnectCB);
+    if (ret != dsERR_NONE) {
+        UT_LOG_ERROR("Failed dsCompositeInRegisterConnectCB(CBFunc:[0x%0X]) dsError_t:[%s]",
+                        compositeInConnectCB, ut_control_GetMapString(dsError_mapTable, ret));
+    }
+    else {
+        UT_LOG_INFO("Result dsCompositeInRegisterConnectCB(CBFunc:[0x%0X]) dsError_t:[%s]",
+                        compositeInConnectCB, ut_control_GetMapString(dsError_mapTable, ret));
+    }
+
+    /* Register Signal change callback */
+    UT_LOG_INFO("Calling dsCompositeInRegisterSignalChangeCB(cbFun:[0x%0X])", compositeInSignalChangeCB);
+    ret = dsCompositeInRegisterSignalChangeCB(compositeInSignalChangeCB);
+    if (ret != dsERR_NONE) {
+        UT_LOG_ERROR("Failed dsCompositeInRegisterSignalChangeCB(cbFun:[0x%0X]) dsError_t:[%s]",
```

Comment:
Consider using ASSERT. Change the print formats. Refer https://github.com/rdkcentral/rdk-halif-test-device_settings/pull/116. Please change through out the function

Reply:
done